imap/jmap_contact.c

```diff
@@ -1631,12 +1631,16 @@ static json_t *jmap_contact_from_vcard(const char *userid,
                 }
                 else if (!strcasecmp(param->name, "label")) {
                     label = param->value;
-                    label_len = strlen(label);
                 }
             }
             json_object_set_new(item, "type", json_string(type));
-            json_object_set_new(item, "label",
-                                label ? json_stringn(label, label_len) : json_null());
+            if (label) {
```

Comment:
Couldn't these repetitive label and label length checks be moved up to, say, line 1612 and initialize the label and label_len once there?